client/src/scenes/Question/scenes/Read/components/FlagsDropdown/FlagsDropdown.jsx

```diff
@@ -8,12 +8,29 @@ import Dropdown, { DropdownItem } from 'components/Dropdown'
 import Button from 'components/Button'
 
 import './FlagsDropdown.css'
+import { useUser } from 'contexts'
 
-const FlagsDropdown = ({ flags, onSelect, onRemove }) => {
+const FlagsDropdown = ({ flags, tags, answer, onSelect, onRemove }) => {
   const intl = getIntl(FlagsDropdown)
   const flagIntl = getIntl(Flag)
 
-  const flagTypes = ['incomplete', 'outdated', 'duplicate']
+  const [flagTypes, setFlagTypes] = useState(['incomplete', 'outdated', 'duplicate'])
+  const { specialities } = useUser()
+
+  useEffect(() => {
+    const updatedFlagTypes = [...flagTypes]
+    if (answer) {
+      tags.forEach(tag => {
+        specialities.forEach(speciality => {
+          if (tag.label.name === speciality.name) {
+            updatedFlagTypes.push('certified')
+          }
+        })
+      })
+    }
+    setFlagTypes(updatedFlagTypes)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [tags, answer])
```

Comment:
Tu n'as pas besoin de `useState` ou `useEffect` ici, la liste des flags à afficher dépend entièrement de `answer`, `tags` (des props) et `specialities` (un contexte).

```suggestion
  const { specialities } = useUser()
  const flagTypes = ['incomplete', 'outdated', 'duplicate']
  
  if (answer) {
    tags.forEach(tag => {
      specialities.forEach(speciality => {
        if (tag.label.name === speciality.name) {
          flagTypes.push('certified')
        }
      })
    })
  }
```

Reply:
J'étais parti sur une solution avec un push au début mais je n'avais pas de fonction pour mettre le if (et vu que je croyais que ce n'était pas recommandé de mettre un if/else hors une fonction) j'ai donc fait un useState / useEffect